File: app/src/main/java/com/example/tutushubham/publicledger/MainActivity.java
package com.example.tutushubham.publicledger;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

public class MainActivity extends AppCompatActivity {

    int PICK_CONTACT;

    private SectionsPagerAdapter mSectionsPagerAdapter;
    private ViewPager mViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());

        mViewPager = findViewById(R.id.container);
        mViewPager.setAdapter(mSectionsPagerAdapter);

        TabLayout tabLayout = findViewById(R.id.tabs);

        mViewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(mViewPager));

        /*FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(Intent.ACTION_PICK,  ContactsContract.Contacts.CONTENT_URI);


                startActivityForResult(intent, 1);
            }
        });*/

    }

    /*@Override
    public void onActivityResult(int reqCode, int resultCode, Intent data) {
        super.onActivityResult(reqCode, resultCode, data);

        switch (reqCode) {
            case 1:
                if (resultCode == Activity.RESULT_OK) {
                    Uri contactData = data.getData();
                    Cursor c =  managedQuery(contactData, null, null, null, null);
                    if (c.moveToFirst()) {
                        String name = c.getString(c.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
                        // TODO Fetch other Contact details as you want to use
                        Toast.makeText(MainActivity.this, name,
                                Toast.LENGTH_LONG).show();
                    }
                }
                break;
        }
    }*/


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public class SectionsPagerAdapter extends FragmentPagerAdapter {

        public SectionsPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {                             //tabs set karna

            switch (position) {
                case 0:
                    Tab1Accounts tab1 = new Tab1Accounts();
                    return tab1;
                case 1:
                    Tab2Calendar tab2 = new Tab2Calendar();
                    return tab2;
                case 2:
                    Tab3Notes tab3 = new Tab3Notes();
                    return tab3;
                case 3:
                    Tab4Calc tab4 = new Tab4Calc();
                    return tab4;
                default:
                    return null;
            }
        }

        @Override
        public int getCount() {           //total tabs
            return 4;
        }

        @Override
        public CharSequence getPageTitle(int position) {            //tabs ka title (buggy)
            switch (position) {
                case 0:
                    return "Accounts";
                case 1:
                    return "Calendar";
                case 2:
                    return "Notes";
                case 3:
                    return "Calculator";
                default:
                    return null;
            }
        }
    }
}

File: app/src/main/java/com/example/tutushubham/publicledger/MainActivity.kt
package com.example.tutushubham.publicledger

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.TabLayoutOnPageChangeListener

class MainActivity : AppCompatActivity() {
    var PICK_CONTACT = 0

    private var mSectionsPagerAdapter: SectionsPagerAdapter? = null
    private var mViewPager: ViewPager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        mSectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager)
        mViewPager = findViewById(R.id.container)
        val tabLayout = findViewById<TabLayout>(R.id.tabs)

        mViewPager?.apply {
            adapter = mSectionsPagerAdapter
            addOnPageChangeListener(TabLayoutOnPageChangeListener(tabLayout))
            tabLayout?.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(this))
        }

        /*FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(Intent.ACTION_PICK,  ContactsContract.Contacts.CONTENT_URI);


                startActivityForResult(intent, 1);
            }
        });*/
    }

    /*@Override
    public void onActivityResult(int reqCode, int resultCode, Intent data) {
        super.onActivityResult(reqCode, resultCode, data);

        switch (reqCode) {
            case 1:
                if (resultCode == Activity.RESULT_OK) {
                    Uri contactData = data.getData();
                    Cursor c =  managedQuery(contactData, null, null, null, null);
                    if (c.moveToFirst()) {
                        String name = c.getString(c.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
                        // TODO Fetch other Contact details as you want to use
                        Toast.makeText(MainActivity.this, name,
                                Toast.LENGTH_LONG).show();
                    }
                }
                break;
        }
    }*/
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    inner class SectionsPagerAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm!!) {
        override fun getItem(position: Int): Fragment {                             //tabs set karna
            return when (position) {
                0 -> {
                    Tab1Accounts()
                }
                1 -> {
                    Tab2Calendar()
                }
                2 -> {
                    Tab3Notes()
                }
                3 -> {
                    Tab4Calc()
                }
                else -> Tab1Accounts() // Default
            }
        }

        override fun getCount(): Int {           //total tabs
            return 4
        }

        override fun getPageTitle(position: Int): CharSequence? {            //tabs ka title (buggy)
            return when (position) {
                0 -> getString(R.string.accounts)
                1 -> getString(R.string.calendar)
                2 -> getString(R.string.notes)
                3 -> getString(R.string.calculator)
                else -> null
            }
        }
    }
}